GUI/QtQuick/QtQuickAdapter.cpp

```diff
@@ -1439,6 +1439,8 @@ bool QtQuickAdapter::validateAddress(const QString& addr)
    const auto& addrStr = addr.toStdString();
    try {
       bs::Address::fromAddressString(addrStr);
+      return QRegularExpression(QString::fromLatin1("^(?:[13]{1}[a-km-zA-HJ-NP-Z1-9]{26,33}|bc1[a-z0-9]{39,59}|tb1[a-z0-9]{39,59})$")
+        ).match(addr).hasMatch();
```

Comment:
Address is parsed in `fromAddressString`, based on the current mainnet/testnet selection, throwing an exception if parsing fails, you don't need to apply a regex on it.